### PR TITLE
schnorrsig

### PR DESCRIPTION
**Pull request by scgbckbone**
* schnorr sign/verify
* xonly pubkey parse/serialize
* bumped libsecp256k1 to newest ~~master 751c4354d51fb5b10a80764df627b84e1a5ccd4f~~ released version `0.2.0`

### CODE REVIEW

**Comment by switck** [comment]
much new code, thanks... but not today

**Comment by scgbckbone** [comment]
no need to hurry - I'd say this needs a thorough review - would really appreciate if you could find some time for it

**Comment by scgbckbone** [comment]
rebased on the top of current master

**Comment by scgbckbone** [comment]
bumped `secp256k1` to released version `0.2.0`

File: ngu/lib_secp256k1.c [hunk 1]
```diff
@@ -1,12 +1,12 @@
 #ifndef NO_QSTR
 
-# ifndef HAVE_CONFIG_H
```
Comment (scgbckbone) [inline ngu/lib_secp256k1.c:3]: configuration header was removed in `v0.3.0` https://github.com/bitcoin-core/secp256k1/blob/master/CHANGELOG.md#removed

File: ngu/lib_secp256k1.c [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
-#  define HAVE_CONFIG_H
-# endif
-
 # define USE_EXTERNAL_DEFAULT_CALLBACKS
 
 # include "src/secp256k1.c"
+# include "src/modules/extrakeys/main_impl.h"
```
Comment (scgbckbone) [inline ngu/lib_secp256k1.c:6]: not sure if this is the best way to do this @doc-hex @switck ? (cope with configuration header removal)

**Review by scgbckbone — COMMENTED** [review]
last commit bumps to newest released version `v0.3.0`

File: ngu/lib_secp256k1.c (hunk 2 above)
Comment (switck) [reply ngu/lib_secp256k1.c:6]: ¯\_(ツ)_/¯

**Pull request merged**